heka/files/toml/decoder/sandbox.toml

```diff
@@ -0,0 +1,63 @@
+[{{ decoder_name }}_decoder]
+type = "SandboxDecoder"
+filename = "{{ decoder.module_file }}"
+
+{% if decoder.module_directory is defined %}
+module_directory = "{{ decoder.module_dir }}"
+{%- endif %}
+
+{% if decoder.memory_limit is defined %}
+memory_limit = "{{ decoder.memory_limit }}"
+{%- endif %}
+
+{% if decoder.preserve_data is defined %}
+preserve_data = "{{ decoder.preserve_data }}"
+{%- endif %}
+
+{% if decoder.instruction_limit is defined %}
+instruction_limit = "{{ decoder.instruction_limit }}"
+{%- endif %}
+
+{% if decoder.output_limit is defined %}
+output_limit = "{{ decoder.output_limit }}"
+{%- endif %}
+
+[{{ decoder_name }}_decoder.config]
```

Comment:
Hmm, I would do it differently. A Lua sandbox plugin (decoder, filter, encoder) is able to pass anything in the [<plugin>.config] section. From the Heka documentation:

> config (object): A map of configuration variables available to the sandbox via read_config. The map consists of a string key with: string, bool, int64, or float64 values.

This is how we've done in the Puppet module => https://github.com/openstack/fuel-plugin-lma-collector/blob/master/deployment_scripts/puppet/modules/heka/templates/decoder/sandbox.toml.erb

I'd imagine that it is possible to do the same in Jinja2 style with Salt passing a dict in the "decoder.config" key. The same applies for the filter and encoder templates.